src/gluonts/dataset/parallelized_loader.py

```diff
@@ -167,28 +168,84 @@ def _as_in_context(batch: dict, ctx: mx.Context):
     return batch
 
 
+def _sequential_sample_generator(
+    dataset: Dataset,
+    transformation: Transformation,
+    is_train: bool,
+    cyclic: bool,
+    num_batches_for_shuffling: int,
+) -> Iterator[DataEntry]:
+    # Approximate shuffling `num_batches_for_shuffling` probabilistically
+    def skip_data_iter(data: Dataset):
+        for data_entry in data:
+            if (
+                random.randint(1, num_batches_for_shuffling)
+                == num_batches_for_shuffling
+            ):
+                yield data_entry
```

Comment:
`p=1/num_batches_for_shuffling ` which leads to `num_batches_for_shuffling` as many time series being considered for a single batch, which is approximately like taking `num_batches_for_shuffling * batch_size` elements, shuffling them, and then returning `batch_size` chunks.

Reply:
This method is much much faster than the alternative described above, essentially because one does not need to store temporarily batches, and because if one batch is requested only one is processed and not `num_batches_for_shuffling` times as many.
The difference is like a 50% increase in performance in mp.